### DISCUSSION TITLE
Implement numeric interface to finally allow filtering for numbers.

### DISCUSSION THREAD

**Discussion by Zxynine** [dotnet/csharplang#6021] [Language Ideas] [answered]
I have spent so long looking at so much code that implements their own hashmap just to find if a generic type is a number or not that its starting to irk me. What we honestly need is some native implementation that allows for us to implement a type constraint specifically for groups of types. If I know as much as I think I do about C#, then it should be relatively easy but completely invaluable to all of us. 

If you create a couple simple interfaces such as INeumeric, IIntegral : INeumeric, IFloatingPoint : INeumeric, and have the number types all inherit from them, we can finally implement generic classes/function that can have a type constraint for numbers! It would be so nice to finally get rid of that nasty boilerplate code many of us have. Preferentially, the interfaces should also force the number types to implement Mult, Div, Add, Sub functions that just utilise their built in operators so that we can do very useful calculations using the generic numbers.

**Comment by HaloFour** [comment, marked as answer]
See: https://devblogs.microsoft.com/dotnet/preview-features-in-net-6-generic-math/

**Comment by orthoxerox** [comment]
https://docs.microsoft.com/en-us/dotnet/api/system.inumber-1?view=net-7.0
